collect_app/src/main/java/org/odk/collect/android/fragments/BarCodeScannerFragment.kt

```diff
@@ -105,24 +97,52 @@ abstract class BarCodeScannerFragment : Fragment() {
                 }
             }
 
-        barcodeScannerViewContainer.barcodeScannerView.start()
-        return rootView
+        binding.barcodeView.barcodeScannerView.start()
+        return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        val binding = FragmentScanBinding.bind(view)
+
         // Layout the prompt/flashlight button under the view finder
         view.addOnLayoutChangeListener { view, _, _, _, _, _, _, _, _ ->
-            val (offset, size) = calculateViewFinder(view.width.toFloat(), view.height.toFloat())
+            val (offset, size) = calculateViewFinder(
+                view.width.toFloat(),
+                view.height.toFloat(),
+                false // We hide these views in landscape/full screen mode
+            )
             val bottomOfViewFinder = offset.y + size.height
 
-            val promptView = view.findViewById<TextView>(R.id.prompt)
+            val promptView = binding.prompt
             val promptLayoutParams = promptView.layoutParams as ConstraintLayout.LayoutParams
             val standardMargin =
                 resources.getDimension(org.odk.collect.androidshared.R.dimen.margin_standard)
             promptView.layoutParams = ConstraintLayout.LayoutParams(promptLayoutParams).also {
                 it.topMargin = (bottomOfViewFinder + standardMargin).toInt()
             }
         }
+
+        updateConfiguration(resources.configuration)
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        updateConfiguration(newConfig)
+    }
+
+    private fun updateConfiguration(config: Configuration) {
+        val binding = FragmentScanBinding.bind(requireView())
+
+        val isLandscape = config.orientation == Configuration.ORIENTATION_LANDSCAPE
+        binding.barcodeView.barcodeScannerView.setFullScreenViewFinder(isLandscape)
+
+        if (isLandscape) {
```

Comment:
This could be simplified:

```
binding.prompt.isVisible = !isLandscape
binding.switchFlashlight.isVisible = !isLandscape
```

Reply:
Agreed! I'm actually converting this to Compose in the follow-up to add the button (now I've been able to discuss design) so let's not bother making this improvement - it'll get wiped out any way.